src/components/Execution/SingleNodeAggregation.jsx

```diff
@@ -1,25 +1,35 @@
 import { FormControlLabel, Switch } from '@material-ui/core'
-import { queryConnect, useClient } from 'cozy-client'
+import { useClient, useQuery } from 'cozy-client'
 import Button from 'cozy-ui/react/Button'
 import Input from 'cozy-ui/react/Input'
 import Label from 'cozy-ui/react/Label'
 import Spinner from 'cozy-ui/react/Spinner'
 import SelectBox from 'cozy-ui/transpiled/react/SelectBox/SelectBox'
 import React, { useCallback, useState } from 'react'
 import { v4 as uuid } from 'uuid'
-import { nodesQuery } from '../../doctypes'
 import createTree from '../../lib/createTreeExported.js'
+import { nodesQuery } from '../../lib/queries'
+import { useEffect } from 'react'
 
-const SingleNodeAggregation = ({ nodes }) => {
+const SingleNodeAggregation = () => {
   const client = useClient()
-
-  const { isLoading, data } = nodes
-  const options = data.map(e => ({ value: e, label: e.label || e.id }))
-
-  const [isWorking, setIsWorking] = useState(false)
+  const query = nodesQuery()
+  const { fetch, isLoading } = useQuery(query.definition, query.options)
+  const [nodes, setNodes] = useState()
   const [nbShares, setNbShares] = useState(3)
   const [pretrained, setPretrained] = useState(true)
-  const [node, setSingleNode] = useState(data[0])
+  const [node, setSingleNode] = useState()
+  const [isWorking, setIsWorking] = useState(false)
+
+  useEffect(() => {
```

Comment:
As usual in those very non satisfying cases: could you add a FIXME comment saying that it shouln't be necessary if `useQuery` was properly working?